mysql-test/suite/rpl/t/rpl_show_slave_hosts.test

```diff
@@ -49,4 +49,173 @@ source include/wait_show_condition.inc;
+source include/wait_for_status_var.inc;
+show status like 'Rpl_semi_sync_master_clients';
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT $SLAVE_MYPORT2 SLAVE_PORT2
+SHOW REPLICA HOSTS;
+
+--echo # Replica 1 changed rpl_semi_sync_enabled value to 0 (asynchronous replication)
+connection slave2;
+stop slave;
+reset slave;
+set global rpl_semi_sync_slave_enabled = 0;
+start slave;
+echo [ slave status should be OFF ];
+show status like 'Rpl_semi_sync_slave_status';
+
+connection master;
+let $status_var= Rpl_semi_sync_master_clients;
+let $status_var_value= 1;
+source include/wait_for_status_var.inc;
+show status like 'Rpl_semi_sync_master_clients';
+echo [ master status should be ON ];
+let $status_var= Rpl_semi_sync_master_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+show status like 'Rpl_semi_sync_master_status';
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT $SLAVE_MYPORT2 SLAVE_PORT2
+SHOW REPLICA HOSTS;
+
+--echo # Stop replica 1 and see the changes 
+connection slave2;
+stop slave;
+connection master;
+let $status_var= Rpl_semi_sync_master_clients;
+let $status_var_value= 1;
+source include/wait_for_status_var.inc;
+show status like 'Rpl_semi_sync_master_clients';
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT $SLAVE_MYPORT2 SLAVE_PORT2
+SHOW REPLICA HOSTS;
+
+--echo #  Replica 2 changed rpl_semi_sync_enabled value to 0
+connection slave;
+stop slave;
+reset slave;
+set global rpl_semi_sync_slave_enabled = 0;
+start slave;
+connection master;
+let $status_var= Rpl_semi_sync_master_clients;
+let $status_var_value= 0;
+source include/wait_for_status_var.inc;
+show status like 'Rpl_semi_sync_master_clients';
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT $SLAVE_MYPORT2 SLAVE_PORT2
+SHOW REPLICA HOSTS;
+
+--echo # Stop replica 2 and see the changes
+connection slave;
+stop slave;
+connection master;
+let $status_var= Rpl_semi_sync_master_clients;
+let $status_var_value= 0;
+source include/wait_for_status_var.inc;
+show status like 'Rpl_semi_sync_master_clients';
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT $SLAVE_MYPORT2 SLAVE_PORT2
+SHOW REPLICA HOSTS;
+
+--echo # Cleanup
+connection slave2;
+start slave;
+connection slave;
+start slave;
+
+connection master;
+DROP TABLE t;
+sync_slave_with_master;
+
+connection master;
+DROP TABLE t1;
+sync_slave_with_master;
+
+--echo # Disconnect slave from asynchronous replication
+connection slave2;
+source include/stop_slave.inc;
+set global rpl_semi_sync_slave_enabled= 0;
+change master to master_user='root',master_password='';
+connection master;
+disconnect slave2;
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT $SLAVE_MYPORT2 SLAVE_PORT2
+SHOW REPLICA HOSTS;
+
+connection master;
+set global rpl_semi_sync_master_enabled = 0;
+set global rpl_semi_sync_master_timeout= default;
```

Comment:
It would be better to restore the server to its state at the start of the test, rather than hard-coded 0 and default (similar to `$save_wait_point`).